.github/workflows/build-push.yml

```diff
@@ -0,0 +1,110 @@
+
+name: Build and Push Docker Image WEAC Codebase
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'python-flask/**'
+      - 'c-sharp-razor/**'
+      - 'rust-actix/**'
+      - 'ruby-sinatra/**'
+      - 'javascript-express/**'
+      - 'go-gin/**'
+      - 'go-gorilla/**'
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        service:
+          - python-flask
+          - c-sharp-razor
+          - rust-actix
+          - ruby-sinatra
+          - javascript-express
+          - go-gin
+          - go-gorilla
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: '1'
+
+      - name: Get Changed Files
+        id: changed-files
+        uses: tj-actions/changed-files@v34
+        with:
+          files: |
+            python-flask/**
+            c-sharp-razor/**
+            rust-actix/**
+            ruby-sinatra/**
+            javascript-express/**
+            go-gin/**
+            go-gorilla/**
+
+      - name: Determine if Service Needs Build
+        id: check_changes
+        run: |
+          if [[ "${{ steps.changed-files.outputs.modified_files }}" == *"${{ matrix.service }}/"* ]] || [[ "${{ steps.changed-files.outputs.added_files }}" == *"${{ matrix.service }}/"* ]]; then
+            echo "Changes detected in ${{ matrix.service }}."
+            echo "service_changed=true" >> $GITHUB_ENV
+          else
+            echo "No changes detected in ${{ matrix.service }}."
+            echo "service_changed=false" >> $GITHUB_ENV
+          fi
+
+      - name: Set up Docker Buildx
+        if: env.service_changed == 'true'
+        uses: docker/setup-buildx-action@v1
+
+      - name: Log in to Docker Hub
+        if: env.service_changed == 'true'
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Check if Dockerfile exists
+        if: env.service_changed == 'true'
+        id: check_file
+        run: |
+          if [ -d "./${{ matrix.service }}" ] && [ -f "./${{ matrix.service }}/docker-compose.yml" ]; then
+            echo "found=true" >> $GITHUB_ENV
+          else
+            echo "found=false" >> $GITHUB_ENV
+          fi
+
+      - name: Install Docker Compose
+        if: env.found == 'true'
+        run: |
+          curl -L "https://github.com/docker/compose/releases/download/1.29.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+          chmod +x /usr/local/bin/docker-compose
+
+      - name: Build and Push Docker Image
+        if: env.found == 'true'
+        run: |
+          cd ${{ matrix.service }}
+          docker-compose build
```

Comment:
why are you building with docker compose instad with the docker file?

Reply:
Same response as above. I thought we are using the `compose-prod` now to handle the environment variables from production